### PR TITLE
Keep "model" in URL when creating fresh model

### PR DESCRIPTION
**Pull request by gusaiani**
### How to Test

1. `+ New`
2. Model
3. Choose either mode

URL should be http://localhost:3000/model/query#…aHash

(before it was http://localhost:3000/*question*/query#…aHash)

### CODE REVIEW

**Review by ranquild — CHANGES_REQUESTED** [review]
I don't think that's the correct way of doing this. We should keep `dataset` flag in the serialized card state and generate a dataset card url when clicking on New -> Model. It seems that we currently don't serialize this flag https://github.com/metabase/metabase/blob/ab828e7b40dbc93316b5c32f5441bd4fc56f6e67/frontend/src/metabase/lib/card.js#L40 so this should be fixed first

**Review by daltojohnso — CHANGES_REQUESTED** [review]

**Comment by gusaiani** [comment]
@ranquild @daltojohnso Now passing dataset from the moment a link to a new model page is serialized.

File: frontend/src/metabase-lib/lib/Question.ts [hunk 1]
```diff
@@ -1270,6 +1274,7 @@ class QuestionInner {
       dataset_query: query.datasetQuery(),
       display: this._card.display,
       parameters: this._card.parameters,
+      dataset,
```
[the remark this answers is not in the document]
Comment (ranquild) [reply frontend/src/metabase-lib/lib/Question.ts:1277]: I believe it the question was the url was created correctly, we wouldn't `dataset` to `serializeUrl`

**Review by ranquild — CHANGES_REQUESTED** [review]
Please take a look at the comment and this loom https://www.loom.com/share/75ee5477af324c9fb5dff3713126100d. Good news is that it seems the approach is correct now

**Comment by gusaiani** [comment]
@ranquild Thanks, it came down do that one-liner.

**Review by ranquild — APPROVED** [review]

**Comment by ranquild** [comment]
@daltojohnso Please take a look to - @gusaiani won't be able to merge the PR since changes were requested

**Review by daltojohnso — APPROVED** [review]

**Pull request merged**